### PR TITLE
Add `rerunInterval` option

### LINKED ISSUE #54: Support the `rerun` parameter
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/alfy/issues/54)

https://www.alfredapp.com/help/workflows/inputs/script-filter/json/

// @mdeboer

> There is a $20.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/alfy/issues/54).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/alfy/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by jopemachine**
## Related Issue
Fixes #54

## Usage

```
alfy.output([{
      title : "",
      arg: "",
      autocomplete: "",
      subtitle: "",
}], 1);
```

## Comment
I made this PR because I think rerun property would be useful.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Please put some more effort into this.

**Comment by sindresorhus** [comment]
This will also need to be well documented in the readme and index.d.ts.

**Comment by sindresorhus** [comment]
A boolean argument will not be very readable at the callsite.

**Comment by jopemachine** [comment]
Thanks for your quick reviewing!

> A boolean argument will not be very readable at the callsite.

I guess maybe there is some misunderstanding.

This is not boolean argument.

This is an interval value from run to next run. (entering the value of the run)

So, The output values for the functions below are as follows.

```
alfy.output([{
      title : "",
      arg: "",
      autocomplete: "",
      subtitle: "",
}], 3);
```

Result: 
```
{
	"items": [
	{
			"title": "",
			"arg": "",
			"autocomplete": "",
			"subtitle": ""
	}],
	"rerun": 3
}

```

Shall I write some documentation about this function?

**Comment by jopemachine** [comment]
> This will also need to be well documented in the readme and index.d.ts.

May I ask where is index.d.ts file?

It seems that I can't find index.d.ts in this repository.

**Comment by sindresorhus** [comment]
> This is not boolean argument.

Just a typo. I meant number. My argument still stands. Not very readable. Should be an options-object.

**Comment by sindresorhus** [comment]
> May I ask where is index.d.ts file?

Ah, I thought we had an index.d.ts here. Never mind.

**Comment by jopemachine** [comment]
```
    alfy.output([{
      title : "",
      arg: "",
      autocomplete: '',
      subtitle: "",
    }], { rerun : 1 });
```

how about this one?

I think It is easy to read and can respond when other options should be added.

**Comment by jopemachine** [comment]
I think it would be good to add `variables` in the same way.

## Usage

```
    alfy.output([{
      title : "",
      arg: "",
      autocomplete: "",
      subtitle: "",
    }], { 
      rerun : 1,
      variables: {
        "fruit": "banana",
        "vegetable": "carrot",
      }
    });
```
Could I add this option to this PR?

**Comment by sindresorhus** [comment]
Try to look at `{ rerun : 1 }` without knowing anything about how it works. Is it clear enough what it does? I don't think so. It needs a clearer name.

**Comment by sindresorhus** [comment]
> I've thought it would be good to add variables in the same way.

That should be a separate PR.

File: readme.md [hunk 1]
```diff
@@ -204,6 +204,52 @@ alfy.output([
 ]);
 ```
 
+##### options - (rerun)
+
+Type: `number`
+
```
Comment (sindresorhus) [inline readme.md:210]: The docs here should link to the Alfred docs about the rerun feature.

**Comment by jopemachine** [comment]
> Try to look at `{ rerun : 1 }` without knowing anything about how it works. Is it clear enough what it does? I don't think so. It needs a clearer name.

So, how about `rerun-interval`?

File: index.js [hunk 2]
```diff
@@ -37,8 +37,10 @@ alfy.alfred = {
 
 alfy.input = process.argv[2];
 
-alfy.output = items => {
-	console.log(JSON.stringify({items}, null, '\t'));
+alfy.output = (items, options) => {
```
Comment (sindresorhus) [inline index.js:40]:
```suggestion
alfy.output = (items, {rerun} = {}) => {
```

File: readme.md (hunk 1 above)
Comment (jopemachine) [reply readme.md:210]: Okay, I will add link, thanks!

File: readme.md [hunk 3]
```diff
@@ -204,6 +204,52 @@ alfy.output([
...
+Re-run the script in given seconds.
```
Comment (sindresorhus) [inline readme.md:211]: This needs a longer and clearer description.

Comment (sindresorhus) [reply readme.md:211]: It should also describe what problem it solves.

**Comment by jopemachine** [comment]
What about the name `run-interval`?

I think It seems to imply that the script will be re-executed based on its interval.

**Comment by jopemachine** [comment]
I've been thinking about the proper variable name.

These are the variable names that I've thought about.

##### 1 - `runInterval` or `rerunInterval`
##### 2 - `outputInterval` or`reOutputInterval`
##### 3 -`updateInterval` or `updateOutputInterval`
##### 4 - `rerunScriptInterval`
##### 5 - `updateEverySpecifiedSecond`

If I still don't have a proper variable name at all, May I ask you to give me some a little hint?

I think it will be very useful if this feature is merged.

**Comment by sindresorhus** [comment]
I would go with `rerunInterval`.

**Comment by sindresorhus** [comment]
Travis is failing

**Comment by jopemachine** [comment]
> Travis is failing

Okay. Let me look into it.

**Comment by jopemachine** [comment]
It was an object-curly-spacing error.

Resolved it by removing the side whitespace of { and }.

**Comment by sindresorhus** [comment]
Since it's difficult to automatically test this, can you manually test that this new feature works?

**Comment by jopemachine** [comment]
> Since it's difficult to automatically test this, can you manually test that this new feature works?

Sure, I tested on my workflow, and I've checked it works.

**Comment by jopemachine** [comment]
> Since it's difficult to automatically test this, can you manually test that this new feature works?

Sure.

Here is my code.

```
  alfy.output([
    {
      title: "Please wait until the caching process is finished...",
      arg: "error",
      autocomplete: "",
      subtitle: `This work could take several minutes.`,
    }
  ], { rerunInterval: 1 });
```
And here is the log.

```
[02:54:36.174] alfred-evernote-workflow[Script Filter] Queuing argument ''
[02:54:36.620] alfred-evernote-workflow[Script Filter] Script with argv '(null)' finished
[02:54:36.623] alfred-evernote-workflow[Script Filter] {
	"items": [
		{
			"title": "Please wait until the caching process is finished...",
			"arg": "error",
			"autocomplete": "",
			"subtitle": "This work could take several minutes."
		}
	],
	"rerun": 1
}
[02:54:37.636] alfred-evernote-workflow[Script Filter] Queuing argument ''
[02:54:38.055] alfred-evernote-workflow[Script Filter] Script with argv '(null)' finished
[02:54:38.057] alfred-evernote-workflow[Script Filter] {
	"items": [
		{
			"title": "Please wait until the caching process is finished...",
			"arg": "error",
			"autocomplete": "",
			"subtitle": "This work could take several minutes."
		}
	],
	"rerun": 1
}
```

**Comment by jopemachine** [comment]
Thank you for merging useful feature.

But, I think there was a mistake in the `README.md` update.

The range of rerun-interval value should be `0.1...5.0` according to [this document.](https://www.alfredapp.com/help/workflows/inputs/script-filter/json/), not to `0.1...0.5`

**Pull request merged**